Automattic/wp-calypso
# Staging Sites: staging sync UI enchancements

Related to https://github.com/Automattic/dotcom-forge/issues/4135

## Proposed Changes

This pr adds various UI enhancements that were left over from the original Sttaging sync prototype.

## Testing Instructions

### Test 1: UI enchancements
1. Create a WoA site.
2. Add a staging site.
3. Navigate to `Hosting` page
4. In your production site ensure that you see the following UI:
![2023-10-10T15:02:47,962693120+03:00](https://github.com/Automattic/wp-calypso/assets/497103/b0027ca4-4695-41ad-afe9-116acfb08ff0)
5. Navigate to your staging site
6. Ensure that you see the following UI:
![2023-10-10T15:02:36,858480603+03:00](https://github.com/Automattic/wp-calypso/assets/497103/0e38a8cb-2241-49f4-9347-b8646b985096)

### Test 2: Selected options order
1. While on either the production or staging site, change the option to sync your changes to production ( you should have some available checkboxes.)
2. Check some of the checkboxes, and press the Synchronize button.
3. Ensure that the options you have selected, appear in the same order as they appear originally.
4. Ensure that the checkboxes and items appear in the same order regardless of the navigation between "pull" and "push" selections.


## Pre-merge Checklist
